src/xaa.ts

```diff
@@ -86,16 +98,16 @@ export class TimeoutError extends Error {
  * It can be an async function.
  * @returns `valOrFunc` or its returned value if it's a function.
  */
-export async function delay(delayMs: number, valOrFunc?: any) {
```

Comment:
I didn't know about the `instanceof Function` trick! This is awesome.

Reply:
I realized later on that typeof should theoretically also narrow.. [Playground Link](https://www.typescriptlang.org/v2/en/play?#code/CYUwxgNghgTiAEYD2A7AzgF3lAXPFArgLYBGIM8APvABQ0Aeehp5AlPALwB88AbkgEtgrANwBYAFCSBAM1oYAngAcQSOVE4cO8AEQyCKMBgGod7AN6T417DSijJAXyA)

I am curious to see why. Can you take a look at my most recent change, which seems to make `typeof` narrow, and see if it makes sense?

My thought is: T can't really be a function, since otherwise it's not possible to distinguish whether to call it or not.